### PR TITLE
Notification Management

### LINKED ISSUE #1: [FEATURE] Add Notification System
**Description**
Adding a notification system to the app would be great to get notified when a new post arrives.

**Tasks**
- [ ] User should get a notification when a new post arrives on the server.

### PR DESCRIPTION
**Pull request by a2ke5e1**
Fixes #1 

Changelog: 
- Change the minimum target SDK to 19 because it is required by the FCM plugin. [536006f](https://github.com/the-hit-times/The-HIT-Times-App-Flutter/pull/3/commits/536006f4bf0e58b9443e4cf572fdd1087ae44446)
- Request permission from the user to display notifications for android 13 and above as well as iOS. [2054aea](https://github.com/the-hit-times/The-HIT-Times-App-Flutter/pull/3/commits/2054aea709e7d9e012e8b1ee81be85debe46f68a)
- Added a dialog box that shows notifications when the app is in the foreground.  [9386a55](https://github.com/the-hit-times/The-HIT-Times-App-Flutter/pull/3/commits/9386a55229a4bf74db6a868cb04149a65f242b93)
- Added Subscription to `posts_notification` for everyone to receive notifications for new posts. [676e4b8](https://github.com/the-hit-times/The-HIT-Times-App-Flutter/pull/3/commits/676e4b8f85b5416a2e680ffdf22fcfbc635c54af)

### CODE REVIEW

**Comment by Ayan-10** [comment]
@a2krocks Provide a little description of what you did in this PR and why you did.

Thanks.

File: lib/notification_service/notification_service.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+
```
Comment (Ayan-10) [inline lib/notification_service/notification_service.dart:5]: Before starting each function, add some comments regarding what you have done inside that function.

File: lib/homepage.dart [hunk 2]
```diff
@@ -21,10 +23,50 @@ class _MainPageState extends State<MainPage> with TickerProviderStateMixin{
   late List<NavigationIconView> _navigationViews;
   late PageController _pageController;
 
+  Future<void> _showMyDialog(RemoteMessage message) async {
+    String? title = message.notification?.title;
+    String? body = message.notification?.body;
+    String? imageUri = message.notification?.android?.imageUrl;
+    // String imageUri = "https://scontent-del1-1.cdninstagram.com/v/t51.2885-15/333940738_737738278003133_5896829832535104130_n.webp?stp=dst-jpg_e35_s480x480&_nc_ht=scontent-del1-1.cdninstagram.com&_nc_cat=105&_nc_ohc=kC3wbot7Q0UAX9I-zZs&edm=ACWDqb8BAAAA&ccb=7-5&ig_cache_key=MzA0OTcwODc5MDc2NjUzMzc3NA%3D%3D.2-ccb7-5&oh=00_AfC89wZFLcT9CQKRFipzWmAL9F3k7iLhAKmN4A2bhsu21w&oe=6407AFDA&_nc_sid=1527a3";
```
Comment (Ayan-10) [inline lib/homepage.dart]: Remove this comment.

**Review by Ayan-10 — CHANGES_REQUESTED** [review]

File: lib/homepage.dart (hunk 2 above)
Comment (a2ke5e1) [reply lib/homepage.dart]: Check it now

**Review by Ayan-10 — APPROVED** [review]

**Pull request merged**